packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -291,13 +294,15 @@ class UpdatePackagesCommand extends FlutterCommand {
           upgrade: true,
           checkLastModified: false,
           offline: offline,
-          flutterRootOverride: temporaryFlutterSdk.path,
+          flutterRootOverride: upgrade
+            ? temporaryFlutterSdk.path
+            : null,
         );
         // Cleanup the temporary SDK
         try {
-          temporaryFlutterSdk.deleteSync(recursive: true);
+          temporaryFlutterSdk?.deleteSync(recursive: true);
         } on FileSystemException {
-          // So sad...
+          // Failed to delete temporary SDK.
```

Comment:
A trace message here that prints the exception object and stack could be helpful.